src/contracts/MockReceiver.sol

```diff
@@ -0,0 +1,244 @@
+
+    address public fraxOft = 0x80Eede496655FB9047dd39d9f418d5483ED600df;
+    address public fraxCurve = 0x53f8F4e154F68C2D29a0D06BD50f82bCf1bd95dB;
+
+    address public sFraxOft = 0x5Bff88cA1442c2496f7E475E9e7786383Bc070c0;
+    address public sFraxCurve = 0xd2866eF5A94E741Ec8EDE5FF8e3A1f9C59c5e298;
+
+    address public frxEthOft = 0x43eDD7f3831b08FE70B7555ddD373C8bF65a9050;
+    address public frxEthCurve = 0x50842664DfBD876249D0113671d72dB168FBE4d0;
+
+    address public sFrxEthOft = 0x3Ec3849C33291a9eF4c5dB86De593EB4A37fDe45;
+    address public sFrxEthCurve = 0xe5F61df936d50302962d5B914537Ff3cB63b3526;
+
+    address public fxsOft = 0x64445f0aecC51E94aD52d8AC56b7190e764E561a;
+    address public fxsCurve = 0xBc383485068Ffd275D7262Bef65005eE7a5A1870;
+
+    address public fpiOft = 0x90581eCa9469D8D7F5D3B60f4715027aDFCf7927;
+    address public fpiCurve = 0x7FaA69f8fEbe38bBfFbAE3252DE7D1491F0c6157;
+
+    /// @dev Initializes the contract.
+    /// @param _endpoint LayerZero Endpoint address
+    constructor(address _endpoint) {
+        endpoint = _endpoint;
+    }
+
+    receive() external payable {}
+
+    /// @dev Using the _oApp address (as provided by the endpoint), return the respective tokens
+    /// @dev ie. a send of FRAX would have the _oApp address of the FRAX OFT
+    /// @return nToken "Native token" (pre-compiled proxy address)
+    /// @return curve (Address of curve.fi pool for nToken/lzToken)
+    function _getRespectiveTokens(address _oApp) internal view returns (address nToken, address curve) {
+        if (_oApp == fraxOft) {
+            nToken = FraxtalL2.FRAX;
+            curve = fraxCurve;
+        } else if (_oApp == sFraxOft) {
+            nToken = FraxtalL2.SFRAX;
+            curve = sFraxCurve;
+        } else if (_oApp == frxEthOft) {
+            nToken = FraxtalL2.WFRXETH;
+            curve = frxEthCurve;
+        } else if (_oApp == sFrxEthOft) {
+            nToken = FraxtalL2.SFRXETH;
+            curve = sFrxEthCurve;
+        } else if (_oApp == fxsOft) {
+            nToken = FraxtalL2.FXS;
+            curve = fxsCurve;
+        } else if (_oApp == fpiOft) {
+            nToken = FraxtalL2.FPI;
+            curve = fpiCurve;
+        } else {
+            revert InvalidOApp();
+        }
+    }
+
+    /// @notice Handles incoming composed messages from LayerZero.
+    /// @dev Decodes the message payload to perform a token swap.
+    ///      This method expects the encoded compose message to contain the swap amount and recipient address.
+    /// @param _oApp The address of the originating OApp.
+    /// @param /*_guid*/ The globally unique identifier of the message (unused in this mock).
+    /// @param _message The encoded message content in the format of the OFTComposeMsgCodec.
+    /// @param /*Executor*/ Executor address (unused in this mock).
+    /// @param /*Executor Data*/ Additional data for checking for a specific executor (unused in this mock).
+    function lzCompose(
```

Comment:
Note that this is the entry point for receiving a bridge tx from another chain.  The process is actually two transactions:
1. OFT is minted to this contract (does not callback).
2. Endpoint calls `lzCompose()` with the composed message sent with the OFT.  At this point, (1) is already executed so the OFT balance should be held within this contract.